File: soal10.py
# Input jumlah buku tersedia
stok_buku = int(input("Masukkan jumlah buku tersedia: "))

# Input jumlah peminjam
jumlah_peminjam = int(input("Masukkan jumlah permintaan peminjaman: "))

antrian = []

# Input nama peminjam
for i in range(jumlah_peminjam):
    nama = input(f"Masukkan nama peminjam ke-{i+1}: ")
    antrian.append(nama)

berhasil = []
gagal = []

# Simulasi proses peminjaman
for nama in antrian:
    if stok_buku > 0:
        berhasil.append(nama)
        stok_buku -= 1
    else:
        gagal.append(nama)  # semua setelah stok habis masuk gagal

# Output hasil
print("\n=== HASIL SIMULASI PEMINJAMAN ===")

print("\nPeminjaman berhasil:")
for nama in berhasil:
    print("-", nama)

print("\nPermintaan peminjaman gagal:")
for nama in gagal:
    print("-", nama)

print("\nSisa buku:", stok_buku)

File: soal10.js
// Jumlah buku tersedia
let stokBuku = 4;

// Daftar permintaan peminjaman (urutan masuk)
const peminjam = ["Alya", "Bima", "Citra", "Doni", "Eka", "Farah"];

// Array hasil
let berhasil = [];
let gagal = [];

// Simulasi proses peminjaman
for (let i = 0; i < peminjam.length; i++) {
  console.log(`Memproses: ${peminjam[i]}`);

  if (stokBuku > 0) {
    berhasil.push(peminjam[i]);
    stokBuku--;
    console.log(`→ Berhasil meminjam buku. Sisa stok: ${stokBuku}\n`);
  } else {
    gagal.push(peminjam[i]);
    console.log(`→ Gagal meminjam (buku habis)\n`);
    break; // proses berhenti
  }
}

// Output akhir
console.log("=== HASIL AKHIR ===");
console.log("Peminjaman Berhasil:");
berhasil.forEach(nama => console.log("- " + nama));

console.log("\nPeminjaman Gagal:");
if (gagal.length > 0) {
  gagal.forEach(nama => console.log("- " + nama));
} else {
  console.log("Tidak ada yang gagal.");
}
